### ISSUE TITLE
HUD background not removed if animation does not complete, makes app unusable until restarted

### ISSUE THREAD

**Issue by ssawchenko** [jdg/MBProgressHUD#380]
https://github.com/jdg/MBProgressHUD/pull/366 already covers the issue I am seeing, but since I could not find a corresponding Issue, I figured I would report this.

In my case I have a tab controller and I can reproduce this by performing an action in one tab that would trigger the HUD and then switching to another tab before the HUD is properly dismissed. On navigating back to the tab with the HUD, the user may no longer interact with the view as the HUD background has not been removed.

This appears to be because removeFromSuperview is not being called because the finished is false.

In doneFinished:

```
    if (finished) {
        self.alpha = 0.0f;
        if (self.removeFromSuperViewOnHide) {
            NSLog(@"HUD removeFromSuperViewOnHide");
            [self removeFromSuperview];
        }
    }
```

This log shows the flow when the tab is switched before the HUD animation can complete:
2016-05-19 14:23:27.396 [9307:4270521] HUD hideAnimated: true
2016-05-19 14:23:27.397 [9307:4270521] HUD hideUsingAnimation started
2016-05-19 14:23:27.400 [9307:4270521] HUD hideUsingAnimation animation completed
2016-05-19 14:23:27.401 [9307:4270521] HUD hideUsingAnimation doneFinished: false

I don't see any benefit to checking for finished here, unless I am missing some functionality of the HUD.  Commenting out the check for finished appears to work for me.

**Comment by zuoming** [comment]
I modify it to :

self.alpha = 0.0f;
    if (self.removeFromSuperViewOnHide) {
            NSLog(@"HUD removeFromSuperViewOnHide");
            [self removeFromSuperview];
        }

**Comment by matej** [comment]
Should be resolved now with #366 merged.
